python/observation-publisher/publisher/observability_backend.py

```diff
@@ -0,0 +1,109 @@
+import abc
+
+import arize
+import pandas as pd
+from arize.pandas.logger import Client, Schema
+from arize.utils.types import Environments
+from merlin.observability.inference import InferenceSchema, InferenceType
+
+from publisher.config import ArizeConfig, ObservabilityBackend, ObservabilityBackendType
+from publisher.prediction_log_parser import PREDICTION_LOG_TIMESTAMP_COLUMN
+
+
+class ObservationSink(abc.ABC):
+    @abc.abstractmethod
+    def write(self, dataframe: pd.DataFrame):
+        raise NotImplementedError
+
+
+def _inference_type_to_arize_model_types(
+    inference_type: InferenceType
+) -> arize.utils.types.ModelTypes:
+    return arize.utils.types.ModelTypes(inference_type.name)
+
+
+class ArizeSink(ObservationSink):
+    def __init__(
+        self,
+        config: ArizeConfig,
+        inference_schema: InferenceSchema,
+        model_id: str,
+        model_version: str,
+    ):
+        self._client = Client(space_key=config.space_key, api_key=config.api_key)
+        # One log will be published per model schema
+        match inference_schema.type:
+            case InferenceType.BINARY_CLASSIFICATION:
+                self._schemas = [
+                    Schema(
+                        feature_column_names=inference_schema.feature_columns,
+                        prediction_label_column_name=inference_schema.binary_classification.prediction_label_column,
+                        prediction_score_column_name=inference_schema.binary_classification.prediction_score_column,
+                        prediction_id_column_name=inference_schema.prediction_id_column,
+                        timestamp_column_name=PREDICTION_LOG_TIMESTAMP_COLUMN,
+                    )
+                ]
+            case InferenceType.MULTICLASS_CLASSIFICATION:
+                self._schemas = [
+                    Schema(
+                        feature_column_names=inference_schema.feature_columns,
+                        prediction_label_column_name=prediction_label_column,
+                        prediction_score_column_name=prediction_score_column,
+                        prediction_id_column_name=inference_schema.prediction_id_column,
+                        timestamp_column_name=PREDICTION_LOG_TIMESTAMP_COLUMN,
+                    )
+                    for prediction_label_column, prediction_score_column in zip(
```

Comment:
usually in multi class the prediction_score_column is multiple depending on the classes but the prediction_label_column must be one usually the label for the highest prediction_score, how will it work with `zip`?

Reply:
since IMO the use case of multiclass is not many currently, I think we can revisit this later need to check how the DS use the multiclass in production

Reply:
so the assumption in here that the prediction_score_column has the same length with the prediction_label_column then, the sample row will look like the on in this https://docs.arize.com/arize/model-types/multiclass-classification#case-2-supports-classification-and-auc-metrics?

Reply:
Yes, that's the expectation.

Reply:
Yes that's the assumption.